spyder/widgets/variableexplorer/dataframeeditor.py

```diff
@@ -615,28 +550,243 @@ def copy(self):
+    """
+    Data Frame level class.
+
+    This class is used to represent index levels in the DataFrameEditor. When
+    using MultiIndex, this model creates labels for the index/header as Index i
+    for each section in the index/header
+
+    Taken from the gtabview project(Level4ExtModel).
+    For more information please see:
+    https://github.com/wavexx/gtabview/blob/master/gtabview/viewer.py
+    """
+
+    def __init__(self, model, palette, font):
+        super(DataFrameLevelModel, self).__init__()
+        self.model = model
+        self._background = palette.dark().color()
+        if self._background.lightness() > 127:
+            self._foreground = palette.text()
+        else:
+            self._foreground = palette.highlightedText()
+        self._palette = palette
+        font.setBold(True)
+        self._font = font
+
+    def rowCount(self, index=None):
+        """Get number of rows (number of levels for the header)."""
+        return max(1, self.model.header_shape[0])
+
+    def columnCount(self, index=None):
+        """Get the number of columns (number of levels for the index)."""
+        return max(1, self.model.header_shape[1])
+
+    def headerData(self, section, orientation, role):
+        """
+        Get the text to put in the header of the levels of the indexes.
+
+        By default it returns 'Index i', where i is the section in the index
+        """
+        if role == Qt.TextAlignmentRole:
+            if orientation == Qt.Horizontal:
+                return Qt.AlignCenter | Qt.AlignBottom
+            else:
+                return Qt.AlignRight | Qt.AlignVCenter
+        if role != Qt.DisplayRole and role != Qt.ToolTipRole:
+            return None
+        if self.model.header_shape[0] <= 1 and orientation == Qt.Horizontal:
+            return 'Index'
+        elif self.model.header_shape[0] <= 1:
+            return None
+        return 'Index ' + to_text_string(section)
+
+    def data(self, index, role):
+        """Get the information of the levels."""
+        if not index.isValid():
+            return None
+        if role == Qt.FontRole:
+            return self._font
+        if index.row() == self.model.header_shape[0] - 1:
+            if role == Qt.DisplayRole:
+                return str(self.model.name(1, index.column()))
+            elif role == Qt.ForegroundRole:
+                return self._foreground
+            elif role == Qt.BackgroundRole:
+                return self._background
+        elif index.column() == self.model.header_shape[1] - 1:
+            if role == Qt.DisplayRole:
+                return str(self.model.name(0, index.row()))
+            elif role == Qt.ForegroundRole:
+                return self._foreground
+            elif role == Qt.BackgroundRole:
+                return self._background
+        elif role == Qt.BackgroundRole:
+            return self._palette.window()
+        return None
+
+
 class DataFrameEditor(QDialog):

```

Comment:
I think large parts of what is added to this class also comes from the `gtabview` project, so acknowledge that here and also at the top of the file.